x/supply/keeper/grpc_query.go

```
@@ -44,3 +44,15 @@
 	}
 	return nil, fmt.Errorf("endpoint disabled")
 }
+
+func (q Querier) TopN(c context.Context, req *types.QueryTopNRequest) (*types.QueryTopNResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	if q.endpointEnabled {
+
+		accounts := q.Keeper.TopN(ctx, q.stakingKeeper.BondDenom(ctx), int(req.N))
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion uint64 -> int

(gosec)